src/librustc_typeck/diagnostics.rs

```diff
@@ -3426,6 +3426,50 @@ parameters. You can read more about it in the API documentation:
 https://doc.rust-lang.org/std/marker/struct.PhantomData.html
 "##,
 
+E0436: r##"
+A struct can include `..` to indicate that you want to use a copy of some other
+struct for some of the values. This functional update syntax can only be used
+on struct types. It is an error to use it on other types, such as enums.
+
+Example of erroneous code:
+
+```compile_fail
+enum Foo {
+    A { x: u32, y: u32 }
+}
+
+fn bar() {
+    baz(Foo::A { x: 5, y: 10 });
+}
+
+fn baz(Foo f) -> Foo {
+    Foo::A { x: 6, ..f } // error: functional record update syntax requires
+                         //        a struct
+}
+```
+
+To fix this, you need to manually initialize the value. For enums, you should match
+on the variants and copy the values to the new enum value:
+
+```
+enum Foo { A { x: u32, y: u32 } }
+
+fn bar() {
+    baz(Foo:A { x: 5, y: 10 });
+}
+
+fn baz(Foo f) -> Foo {
+    match f {
+        // Match each variant of the enum and bind each value to a name.
+        Foo::A { y: y, .. } =>
+            // Create a new value of the same enum type, set the updated value
+            // and the values that were bound.
```

Comment:
@nrc "value" was supposed to refer to the field values the user wanted to change and "values" was supposed to refer to the field values that should be copied from the initial enum value. There's probably a better way to explain that...